program/mavlink/mission.c

```diff
@@ -319,8 +318,6 @@ void mission_write_waypoint_list(void)
 	/* Update the wayppoint, navigation manager */
 	waypoint_info.waypoint_count = new_waypoint_list_count;
 	waypoint_info.is_busy = false;
-	simple_waypoint_have_been_updated = false;
-	nav_waypoint_list_is_updated = false;
 	/* Send a mission ack Message at the end */
```

Comment:
This `nav_waypoint_list_is_updated = false;` do not in your new version. So I remove it.